src/hooks/useAppState/middleware.js

```diff
@@ -16,34 +18,44 @@ const applyMiddleware = (dispatch, getState) => (action) => {
   switch (type) {
     case types.DELETE_ITEM_BY_KEY: {
       const state = getState();
-      const itemToRemove = state.items.find((item) => item.key === data.key);
+      const itemToRemove = selectItems(state).find((item) => item.key === data.key);
```

Comment:
Using these new helper functions to get the items from the state (it's no longer `state.items`)